### PR TITLE
LT-4621-New-Notifications-Implement-on-off-configuration

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/Lykke.Snow.Notifications.Domain/Model/DeviceConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            public NotificationType Type { get; }
+            public bool Enabled { get; }
+        }
+
+        public string DeviceId { get; }
+        public string AccountId { get; }
+        public string Locale { get; set; }
+        public IReadOnlyList<Notification> Notifications { get; }
+
+        public IEnumerable<Notification> EnabledNotifications => Notifications.Where(n => n.Enabled);
```
[the remark this answers is not in the document]
Comment (tarurar) [reply src/Lykke.Snow.Notifications.Domain/Model/DeviceConfiguration.cs:38]: Do you think iе is better to create a hashset every time we check EnabledNotifications collection? I mean, is it still beneficial having HashSet for faster search but creating and instance of HashSet every time?

File: src/Lykke.Snow.Notifications.Domain/Model/DeviceConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            public Notification(string type, bool enabled = true)
+            {
+                if (string.IsNullOrWhiteSpace(type))
+                    throw new ArgumentNullException(nameof(type), "Notification type cannot be null or empty");
+
+                if (!Enum.TryParse<NotificationType>(type, true, out var notificationType))
+                    throw new ArgumentException($"Notification type [{type}] is not supported");
+
+                Type = notificationType;
+                Enabled = enabled;
```
[the remark this answers is not in the document]
Comment (tarurar) [reply src/Lykke.Snow.Notifications.Domain/Model/DeviceConfiguration.cs:26]: Well, no. Enabled can be false, actually. The code will work fine even if we have a database entry with enabled = false. As we already discussed, the approach when you have only enabled entries in a database does not give you a possibility to have additional options/properties for notification type  once it is disabled and removed from the table.

File: src/Lykke.Snow.Notifications.Domain/Model/DeviceConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            }
+
+            public NotificationType Type { get; }
+            public bool Enabled { get; }
+        }
+
+        public string DeviceId { get; }
+        public string AccountId { get; }
+        public string Locale { get; set; }
```
[the remark this answers is not in the document]
Comment (tarurar) [reply src/Lykke.Snow.Notifications.Domain/Model/DeviceConfiguration.cs]: Okay, added locale enum.

File: src/Lykke.Snow.Notifications/Startup/HostConfiguration.cs [hunk 4]
```diff
@@ -24,9 +24,18 @@ public static IHostBuilder ConfigureHost(this WebApplicationBuilder builder, ICo
...
-                    cBuilder.RegisterModule(new DalModule(settings.CurrentValue.NotificationService.Db.ConnectionString));
-                    cBuilder.RegisterModule(new CqrsModule(settings.CurrentValue.NotificationService.Cqrs));
                     cBuilder.RegisterModule(new NotificationsModule(settings.CurrentValue.NotificationService));
+                    
+                    // @atarutin: Due to a known bug in ASP.NET Core since version 3.0 ConfigureTestContainer is not
+                    // being called when using WebApplicationFactory for integration testing, therefore "environment
+                    // name" approach is used here to not register environment-specific modules in test environment.
+                    // Once the bug is fixed ConfigureTestContainer should be used instead
+                    // LINK: https://github.com/dotnet/aspnetcore/issues/14907
+                    if (!ctx.HostingEnvironment.IsEnvironment("test"))
```
Comment (fatihdumanli) [inline src/Lykke.Snow.Notifications/Startup/HostConfiguration.cs:34]: Will our own test environment bypass this? It's worth considering a more specific name like "integration-tests" or something.

Comment (tarurar) [reply src/Lykke.Snow.Notifications/Startup/HostConfiguration.cs:34]: It is just a matter of taste.
The very same environment has to be used in WebApplicationFactory we are using for integration tests: https://github.com/LykkeBusiness/Lykke.Snow.Notifications/blob/b2d365120972ccf367724e7073f9a21264079d73/tests/Lykke.Snow.Notifications.Tests/WebAppFactoryNoDependencies.cs#L19.

Comment (tarurar) [reply src/Lykke.Snow.Notifications/Startup/HostConfiguration.cs]: On Kubernetes there is default `ASPNETCORE_ENVIRONMENT` setup which is Production cause we emulate the customer environment. Yes, if the environment name will be "test" then we are not gonna register data and cqrs modules. Your remark makes sense. Changed the name to "integration-tests".

**Pull request closed without merging**